common/src/main/java/com/radixdlt/monitoring/Metrics.java

```diff
@@ -281,6 +283,26 @@ public MethodId(Class<?> cls, String methodName) {
     }
   }
 
+  public record RoundChange(
+      String leaderKey,
+      String leaderComponentAddress,
```

Comment:
just making sure: in all code paths, these 2 strings can only have some well-bounded cardinality, right? (assuming malicious other nodes)

Reply:
well, not strictly bounded as the validator set can change across epochs, but I don't think it's a concern

Reply:
Thinking about this a little more, perhaps we should split this metric and remove the leader from this to avoid combinatorial explosion possibilities.

And then we can have a separate metric for round results which we pull from the engine like Olympia - that way we don't miss our own round results if we lose connection for a while and ledger-sync our missed round.